qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -793,56 +831,53 @@ def _common_extend_steps(self, md_template):
                 # code). Sorting the new columns to enforce an order
                 new_cols = sorted(new_cols)
 
-                sql_alter = """ALTER TABLE qiita.{0} ADD COLUMN {1} {2}"""
-                for category in new_cols:
-                    qdb.sql_connection.TRN.add(
-                        sql_alter.format(table_name, category, 'varchar'))
+                cols = self.categories()
+                cols.extend(new_cols)
+
+                values = '{"columns": %s}' % dumps(cols)
+                sql = """UPDATE qiita.{0}
+                         SET sample_values = %s
+                         WHERE sample_id = '{1}'""".format(
+                            table_name, QIITA_COLUMN_NAME)
+                qdb.sql_connection.TRN.add(sql, [values])
 
                 if existing_samples:
                     # The values for the new columns are the only ones that get
                     # added to the database. None of the existing values will
                     # be modified (see update for that functionality)
                     md_filtered = md_template[new_cols].loc[existing_samples]
-                    values = [list(md_filtered[h]) for h in md_filtered]
-                    values.append(existing_samples)
-                    values = [list(v) for v in zip(*values)]
-                    # psycopg2 requires a list of iterable, in which each
-                    # iterable is a set of values to use in the string
-                    # formatting of the query. We have all the values in
-                    # different lists (but in the same order) so use zip to
-                    # create the list of iterable that psycopg2 requires.
-                    set_str = ["{0} = %s".format(col) for col in new_cols]
-                    sql = """UPDATE qiita.{0}
-                             SET {1}
-                             WHERE sample_id=%s""".format(table_name,
-                                                          ",".join(set_str))
-                    for v in values:
-                        qdb.sql_connection.TRN.add(sql, v)
+                    for sid, df in md_filtered.iterrows():
+                        values = dict(df.iteritems())
+                        sql = """UPDATE qiita.{0}
+                                 SET sample_values = sample_values || %s
```

Comment:
Similarly, if I read this correctly, it might be good to remind the reader that sample_values is a JSONB column, or maybe we can include a URL reference to JSONB. I can look for one.

Reply:
The JSONB-using queries look slick! After going through the code, I noticed some were easier to identify as JSONB-using (those that use ->>), while others weren't (Those that used the '-' operator). I'd like to include a link to a JSONB reference at the top of the files that use them, or else comment the queries that specifically use them, to make it bonehead easy to identify.

Reply:
same, adding but this is like asking what does && or || in an if ...